### PR TITLE
Add Jenkins test cases

### PR DESCRIPTION
**Pull request by OdedViner**
Signed-off-by: Oded Viner <odedviner@localhost.localdomain>
OCS-2175- Jenkins application: Deployment
OCS-2176- Jenkins application: Master Node drain
OCS-2205- Jenkins application: Restart MGR pod
OCS-2204- Jenkins application: Restart MON pod
OCS-2179- Jenkins application: Restart OSD pod
OCS-2178- Jenkins application: Reboot app node
OCS-2202 - Jenkins application: Reboot Master node

### CODE REVIEW

File: tests/e2e/workloads/jenkins/test_jenkins_node_reboot.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        jenkins.number_builds_per_project = num_of_builds
+
+        # Start Builds
+        jenkins.start_build()
+
+        # Get relevant node
+        node1 = jenkins.get_nodes(node_type=node_type, num_of_nodes=1)
+
+        # Reboot relevant node
+        if len(node1) > 0:
```
Comment (PrasadDesala) [inline tests/e2e/workloads/jenkins/test_jenkins_node_reboot.py:83]: Same comment as in above test

File: ocs_ci/ocs/jenkins.py [hunk 2]
```diff
@@ -288,22 +291,65 @@ def create_ocs_jenkins_template(self):
...
-            ocp_obj = OCP(namespace=build[1], kind='build')
-            output_build = ocp_obj.describe(resource_name=build[0])
-            list_build = output_build.split()
-            index = list_build.index('Duration:')
-            build_table.add_row([build[1], build[0], list_build[index + 1]])
+        for build, time_build in self.build_completed.items():
+            build_table.add_row([build[1], build[0], time_build])
         log.info(f'\n{build_table}\n')
 
+    def get_nodes(self, node_type='worker', num_of_nodes=1):
```
Comment (PrasadDesala) [inline ocs_ci/ocs/jenkins.py:320]: Can you please change the function name to more suitable name

File: ocs_ci/ocs/jenkins.py [hunk 3]
```diff
@@ -288,22 +291,65 @@ def create_ocs_jenkins_template(self):
...
+        if node_type == 'master':
+            nodes1 = []
+            nodes = get_typed_nodes(node_type=node_type, num_of_nodes=num_of_nodes)
+            for node in nodes:
+                nodes1.append(node.name)
+        elif node_type == 'worker':
+            pod_objs = []
+            for project in self.projects:
+                pod_names = get_pod_name_by_pattern(
+                    pattern='jenkins', namespace=project
```
Comment (PrasadDesala) [inline ocs_ci/ocs/jenkins.py:340]: Does all the pods created in the test  has the same pattern 'jenkins' ?
(app pod, build pod etc)

Comment (OdedViner) [reply ocs_ci/ocs/jenkins.py:340]: yes

**Pull request closed without merging**